src/addons/astar.js

```diff
@@ -0,0 +1,117 @@
+import _ from 'lodash';
+import * as d3 from 'd3';
+
+const astar = (G) => {
+  /**
+   * A* super basic path searching, not terribly optimized but fast enough for me
+   *
+   * @param {object} start - coord {x:%, y:%}
+   * @param {string} goal - coord {x:%, y:%}
+   * @param {function} collider - used during pathing, for routing around obstacles
+   * @param {object} grid - optional {x:%, y:%}, default 10,10 - routing is done on this grid
+   * @param {number} searchLimit - this is the max number of grid squares astar will search, default 1000
+   */
+  const getPath = (start, goal, collider = () => false, grid = { x:10, y:10 }, searchLimit = 1000) => {
```

Comment:
Per discussion maybe rename grid => cell or gridCell, makes things a little easier to understand that it is the unit of movement and not the canvas itself as the grid.